ygnmi/types.go

```diff
@@ -22,43 +22,64 @@ import (
 	"github.com/openconfig/ygot/ytypes"
 )
 
-func newBaseQuery[T any](dir string, state bool, ps PathStruct, schemaFn func() *ytypes.Schema) baseQuery[T] {
+// CompressionInfo contains information about a compressed path element for a
+// node which points to a path element that's compressed out.
+//
+// e.g. for OpenConfig's /interfaces/interface, if a path points to
+// /interfaces, then CompressionInfo will be populated with the following:
+// - PreRelPath: []{"openconfig-interfaces:interfaces"}
+// - PostRelPath: []{"openconfig-interfaces:interface"}
+type CompressionInfo struct {
+	// PreRelPath is the list of qualified path elements prior to the
+	// compressed-out node.
+	PreRelPath []string
+	// PostRelPath is the list of qualified path elements after the
+	// compressed-out node.
+	PostRelPath []string
+}
+
+func newBaseQuery[T any](dir string, state bool, ps PathStruct, schemaFn func() *ytypes.Schema, extractFn ExtractFn[T], goStructFn func() ygot.ValidatedGoStruct) baseQuery[T] {
 	return baseQuery[T]{
 		goStructName: dir,
 		state:        state,
 		ps:           ps,
 		yschemaFn:    schemaFn,
+		extractFn:    extractFn,
+		goStructFn:   goStructFn,
 	}
 }
 
 // NewLeafSingletonQuery creates a new LeafSingletonQuery object.
 func NewLeafSingletonQuery[T any](parentDir string, state, scalar bool, ps PathStruct, extractFn ExtractFn[T], goStructFn func() ygot.ValidatedGoStruct, schemaFn func() *ytypes.Schema) *LeafSingletonQuery[T] {
 	return &LeafSingletonQuery[T]{
 		leafBaseQuery: leafBaseQuery[T]{
-			baseQuery: newBaseQuery[T](
+			baseQuery: newBaseQuery(
 				parentDir,
 				state,
 				ps,
 				schemaFn,
+				extractFn,
+				goStructFn,
 			),
-			scalar:     scalar,
-			extractFn:  extractFn,
-			goStructFn: goStructFn,
+			scalar: scalar,
 		},
 	}
 }
 
 // NewNonLeafSingletonQuery creates a new NonLeafSingletonQuery object.
-func NewNonLeafSingletonQuery[T any](dir string, state bool, ps PathStruct, subPaths []PathStruct, schemaFn func() *ytypes.Schema) *NonLeafSingletonQuery[T] {
+func NewNonLeafSingletonQuery[T any](dir string, state bool, ps PathStruct, extractFn ExtractFn[T], goStructFn func() ygot.ValidatedGoStruct, subPaths []PathStruct, schemaFn func() *ytypes.Schema, compressInfo *CompressionInfo) *NonLeafSingletonQuery[T] {
```

Comment:
when aren't these nil?

Reply:
I think this has been resolved, feel free to reopen if there are other comments.